plugins/network-elements/nsx/src/main/java/org/apache/cloudstack/resource/NsxResource.java

```diff
@@ -360,6 +361,8 @@ private NsxAnswer executeRequest(DeleteNsxSegmentCommand cmd) {
         try {
             Thread.sleep(30*1000);
             String segmentName = getSegmentName(cmd);
+            DhcpRelayConfigs dhcpRelayConfig = (DhcpRelayConfigs) nsxService.apply(DhcpRelayConfigs.class);
+            dhcpRelayConfig.delete(getDhcpRelayId(cmd));
```

Comment:
We may need to test if its needed to remove the link from the segment to the DHCP relay before this deletion

Reply:
Ah ok, we can call the delete dhcp relay, after the segment deletion